share/transition_scripts/openldap_to_389ds/openldap_to_389ds.rb

```diff
@@ -0,0 +1,216 @@
+#!/usr/bin/env ruby
+#
+# frozen_string_literal: true
+
+#  This script is used to convert user and group data from a SIMP OpenLDAP
+#  server into a LDIF that can be imported into a SIMP 389-DS server that has
+#  the same base DN.
+#
+#    This is a list of what it changes on the LDIF entries:
+#
+#    - Moves entries from ou=Group,<basedn> to ou=Groups,<basedn>
+#      dsidm, a tool used to maintain entries in 389-DS, is opinionated at
+#      this time on where groups and users should be.  It expects them
+#      to be in  ou=Groups and ou=People.
+#    - Removes attributes that are not compatible with 389-DS (see
+#      `DELETE_KEYS` for a full list)
+#    - Converts SSH  attributes to the format used by 389-DS.
+#    - Adds `objectClass` attributes needed by dsidm.
+#    - Converts `lockout` settings.
+#
+require 'fileutils'
+require 'net-ldap'
+require 'optparse'
+
+# Attributes that will cause issues in 389-DS
+# - memberuid is copied into to member before deletion
+# - sshpublickey is copied into nssshpublickey before deletion
+
+DELETE_KEYS = %i[
+  entrycsn
+  entryuuid
+  memberuid
```

Comment:
In sssd we have the schema set to rfc2307.  This looks for group membership using memberUid.  If we do this then we need to change the schema in the ldap provider to rfc2307bis.      I don't know what other effects this might have.   If we decide to keep it rfc2307 then we need to update the docs in simp-doc.

Reply:
@jeannegreulich :+1: we should definitely change it to `rfc2307bis` in `simp::sssd::client`